contrib/opencensus-ext-azure/opencensus/ext/azure/extension/azure_functions.py

```diff
@@ -0,0 +1,154 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from logging import Logger
+from typing import Dict, List, Optional
+
+from azure.functions import (
+    AppExtensionBase,
+    Context,
+    FunctionExtensionException,
+)
+from opencensus.trace import config_integration
+from opencensus.trace.propagation.trace_context_http_header_format import (
+    TraceContextPropagator,
+)
+from opencensus.trace.samplers import ProbabilitySampler
+from opencensus.trace.tracer import Tracer
+
+from ..trace_exporter import AzureExporter
+
+# The app setting name for providing default opencensusu appinsights key
+DEFAULT_INSTRUMENTATION_KEY_SETTING = 'APPINSIGHTS_INSTRUMENTATIONKEY'
+
+# The instrumentatino key prefix in Azure Monitor connection string
+INSTRUMENTATION_KEY_PREFIX = 'InstrumentationKey='
+
+
+class OpenCensusExtension(AppExtensionBase):
+    """Extension for Azure Functions integration to export traces into Azure
+    Monitor. Ensure the following requirements are met:
+    1. Azure Functions version is greater or equal to v3.0.15584
+    2. App setting PYTHON_ENABLE_WORKER_EXTENSION is set to 1
+    3. OpenCensusExtension.configure() method is called
+    """
+
+    @classmethod
+    def init(cls):
+        cls._tracers: Dict[str, Tracer] = {}
+        cls._exporter: Optional[AzureExporter] = None
+        cls._has_configure_called: bool = False
+        cls._trace_integrations: List[str] = []
+        cls._default_instrumentation_key: Optional[str] = os.environ.get(
+            DEFAULT_INSTRUMENTATION_KEY_SETTING
+        )
+
+    @classmethod
+    def configure(cls,
+                  libraries: List[str],
+                  connection_string: Optional[str] = None,
+                  *args,
+                  **kwargs):
+        """Configure libraries for integrating into OpenCensus extension.
+        Initialize an Azure Exporter that will write traces to AppInsights.
+
+        :type libraries: List[str]
+        :param libraries: the libraries opencensus-ext-* that need to be
+            integrated into OpenCensus tracer. (e.g. ['requests'])
+        :type connection_string: Optional[str]
+        :param connection_string: the connection string for azure exporter
+            to write into. If this is set to None, the extension will write to
+            the AppInsights resource defined in APPINSIGHTS_INSTRUMENTATIONKEY
+        """
+        cls._has_configure_called = True
+
+        cls._trace_integrations = config_integration.trace_integrations(
+            libraries
+        )
+
+        instrumentation_key = cls._parse_ikey_from_cstring(connection_string)
```

Comment:
[AzureExporter](https://github.com/census-instrumentation/opencensus-python/blob/master/contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py#L62) already performs all the validation logic for connection_string/ikey, so you can remove everything that is related to checking from the extension. No need for te parsing logic either. Simply call the constructor of `AzureExporter` like so: `AzureExporter(connection_string=connection_string, instrumentation_key=instrumentation_key)` and the exporter will handle the parsing, prioritization and None checking,